apps/meteor/app/livechat/imports/server/rest/users.js

```diff
@@ -1,17 +1,26 @@
 import { check } from 'meteor/check';
 import _ from 'underscore';
 
-import { hasPermission } from '../../../../authorization';
 import { API } from '../../../../api/server';
 import { Users } from '../../../../models/server';
 import { Livechat } from '../../../server/lib/Livechat';
 import { findAgents, findManagers } from '../../../server/api/lib/users';
+import { hasAtLeastOnePermissionAsync } from '../../../../authorization/server/functions/hasPermission';
 
 API.v1.addRoute(
 	'livechat/users/:type',
-	{ authRequired: true },
 	{
-		get() {
+		authRequired: true,
+		permissionsRequired: {
+			GET: {
+				permissions: ['manage-livechat-agents'],
+				operation: 'hasAll',
+			},
```

Comment:
I believe there's a small issue here which changes the permission checks for this endpoint. If we take an example of agents (i.e. `:type=agent`), right now any user with **one of these 3 permission** is allowed to query this endpoint

- 'manage-livechat-agents'
- 'transfer-livechat-guest'
- 'edit-omnichannel-contact'

But with this new logic, the user will always need to have 'manage-livechat-agents' role & users with who don't have this role but do have the other 2 roles won't be able to access this endpoint